api-registration/server/index.js

```diff
@@ -0,0 +1,85 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  argon2.hash(password).then(hashedPassword => {
+    const sql = `
+            insert into "users"
+              ("username", "hashedPassword")
+              values($1, $2)
+              returning *`;
+    const userAuth = [username, hashedPassword];
+    db.query(sql, userAuth).then(result => {
+      res.status(201).json(result.rows[0]);
+    }).catch(err => next(err));
+
+  }).catch(err => next(err));
+
+  /* your code starts here */
+
+  /**
+   * Hash the user's password with `argon2.hash()`
+   * Then, 😉
+   *   Insert the user's "username" and "hashedPassword" into the "users" table.
+   *   Then, 😉
+   *     Respond to the client with a 201 status code and the new user's "userId", "username", and "createdAt" timestamp.
+   *   Catch any errors.
+   * Catch any errors.
+   *
+   * Hint: Insert statements can include a `returning` clause to retrieve the insterted row(s).
+   */
+
+});
+
+app.delete('/api/auth/:userId', (req, res, next) => {
+  const userId = Number(req.params.userId);
+  if (!Number.isInteger(userId) || userId < 1) {
+    throw new ClientError(400, 'userId must be a positive integer');
+  }
+  const sql = `
```

Comment:
This is all well and good but users can now delete each other.

Reply:
> This is all well and good but users can now delete each other.

![bwahhah](https://user-images.githubusercontent.com/49361894/188750685-d082a53a-3ffb-45eb-b570-3f610198cf0d.jpeg)